didcomm/core/utils.py

```diff
@@ -0,0 +1,134 @@
+import dataclasses
+from typing import Union, Optional
+
+from authlib.common.encoding import to_unicode, urlsafe_b64decode, to_bytes
+from authlib.jose import ECKey, OKPKey
+from authlib.jose.rfc7517 import AsymmetricKey
+
+from didcomm.common.algorithms import SignAlg
+from didcomm.common.types import (
+    VerificationMaterialFormat,
+    VerificationMethodType,
+    DID_OR_DID_URL,
+    DID_URL,
+    DID,
+)
+from didcomm.core.serialization import json_str_to_dict
+from didcomm.did_doc.did_doc import VerificationMethod
+from didcomm.errors import DIDCommValueError
+from didcomm.secrets.secrets_resolver import Secret
+
+
+def extract_key(
+    verification_method: Union[VerificationMethod, Secret]
+) -> AsymmetricKey:
+    if (
+        verification_method.type == VerificationMethodType.JSON_WEB_KEY_2020
+        and verification_method.verification_material.format
+        == VerificationMaterialFormat.JWK
+    ):
+        jwk = json_str_to_dict(verification_method.verification_material.value)
+        if jwk["kty"] == "EC":
+            return ECKey.import_key(jwk)
+        elif jwk["kty"] == "OKP":
+            return OKPKey.import_key(jwk)
+        raise DIDCommValueError()
```

Comment:
not the only place in the repo but an `raise DIDCommValueError` (whish is equal) looks more usual, I believe